### PR TITLE
feat: add support for agent context history

### PR DESCRIPTION
**Pull request by jpvajda**
### TL;DR
Successfully implemented the Function Call Context / History feature for the Deepgram Python SDK, enabling agents to maintain conversation context and function call history across sessions. The implementation is fully API specification compliant and includes comprehensive testing.

### **New Classes Added:**
- **`Flags`**: Configuration flags with `history` boolean (defaults to `true`)
- **`Context`**: Container for conversation history messages
- **`HistoryConversationMessage`**: Conversation text messages in history format
- **`HistoryFunctionCallsMessage`**: Function call messages in history format  
- **`FunctionCallHistory`**: Individual function call records with id, name, arguments, response, and client_side flag

### **Enhanced Existing Classes:**
- **`Agent`**: Added optional `context` field for conversation history
- **`SettingsOptions`**: Added optional `flags` field for feature flags

### **Key Features:**
- ✅ Full API specification compliance
- ✅ Backward compatibility maintained
- ✅ Union type handling for mixed message types
- ✅ Proper serialization/deserialization
- ✅ Comprehensive error handling
- ✅ Developer-friendly construction patterns

### **Unit Tests (41 new tests):**
- `TestFlags`: 6 tests for flags functionality
- `TestHistoryConversationMessage`: 5 tests for conversation messages
- `TestFunctionCallHistory`: 4 tests for function call records
- `TestHistoryFunctionCallsMessage`: 6 tests for function call messages
- `TestContext`: 6 tests for context container
- `TestAgentIntegration`: 3 tests for agent integration
- `TestSettingsOptionsIntegration`: 5 tests for settings integration
- `TestErrorHandling`: 6 tests for edge cases

### **Demo Application:**
- Comprehensive schema testing focused on implementation validation
- Real-world usage patterns demonstration
- Edge case testing
- API specification compliance validation

### **Regression Testing:**
- All 193 existing tests pass (152 original + 41 new)
- No regressions introduced

### **Flags Structure:**
```json
{
  "flags": {
    "history": true
  }
}
```

### **Agent Context Structure:**
```json  
{
  "agent": {
    "context": {
      "messages": [
        {
          "type": "History",
          "role": "user", 
          "content": "What's the weather?"
        },
        {
          "type": "History",
          "function_calls": [
            {
              "id": "fc_12345",
              "name": "get_weather",
              "client_side": true,
              "arguments": "{\"location\": \"NYC\"}",
              "response": "Sunny, 72°F"
            }
          ]
        }
      ]
    }
  }
}
```

## Usage Example

```python
from deepgram.clients.agent import (
    SettingsOptions, Agent, Flags, Context,
    HistoryConversationMessage, HistoryFunctionCallsMessage, 
    FunctionCallHistory
)

# Create conversation history
conversation_msg = HistoryConversationMessage(
    role="user",
    content="What's my account balance?"
)

# Create function call history
function_call = FunctionCallHistory(
    id="fc_balance_123",
    name="get_account_balance", 
    client_side=True,
    arguments='{"account_id": "12345"}',
    response="Current balance: $1,250.50"
)
func_msg = HistoryFunctionCallsMessage(function_calls=[function_call])

# Create context and settings
context = Context(messages=[conversation_msg, func_msg])
settings = SettingsOptions(
    flags=Flags(history=True),
    agent=Agent(
        language="en",
        context=context
    )
)

# Use in agent websocket connection
connection = client.agent.websocket.v("1")
connection.start(settings)
```

### **Core Implementation:**
- `deepgram/clients/agent/v1/websocket/options.py` - Added new classes and enhanced existing ones

### **Export Updates:**
- `deepgram/clients/agent/v1/websocket/__init__.py`
- `deepgram/clients/agent/v1/__init__.py` 
- `deepgram/clients/agent/client.py`
- `deepgram/clients/agent/__init__.py`

### **Testing:**
- `tests/unit_test/test_unit_agent_history_context.py` - Comprehensive test suite

### **Demo (temporary):**
- `temp-test/demo_history_context.py` - Schema validation demo (cleaned up)

### **Code Quality:**
- ✅ Follows existing SDK patterns and conventions
- ✅ Proper docstrings and type hints
- ✅ Consistent error handling
- ✅ No linter errors or warnings

### **Architecture:**
- ✅ Maintains backward compatibility  
- ✅ Uses standard `dataclasses-json` patterns
- ✅ Proper Union type handling with `__post_init__` discrimination
- ✅ Clean separation of concerns

### **Testing Coverage:**
- ✅ 41 comprehensive unit tests added
- ✅ All edge cases covered
- ✅ Schema validation included
- ✅ Integration tests with existing functionality
- ✅ No regressions (193/193 tests passing)

### **Documentation:**
- ✅ Comprehensive usage examples
- ✅ Clear API specification mapping
- ✅ Realistic developer usage patterns
- ✅ Edge case documentation

---

**🚀 Ready for Production**: The Function Call Context / History feature is fully implemented, tested, and ready for release!
## Types of changes

What types of changes does your code introduce to the community Python SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jpvajda** [comment]
Use this preview build URL: https://deepgram-preview-fffa92b2-f434-4c0a-80a0-98427d2dbc55.docs.buildwithfern.com/home

the one above is wonky.

**Comment by jpvajda** [comment]
I need to double checkout the approach after chatting with @lukeocodes

**Comment by jpvajda** [comment]
I validated this approach is correct, but this PR will need a rebase once we merge in #562  which I can do when it's merged, else it's ready for review.

File: deepgram/clients/agent/v1/websocket/options.py [hunk 1]
```diff
@@ -244,6 +244,94 @@ def __getitem__(self, key):
...
+        if not self.messages:
+            self.messages = []
+            return
+
+        # Convert dicts to appropriate message objects and filter out None values
+        converted_messages = []
+        for message in self.messages:
+            if message is None:
+                continue
+            elif isinstance(message, dict):
```
Comment (naomi-lgbt) [inline deepgram/clients/agent/v1/websocket/options.py]:
```suggestion
            if isinstance(message, dict):
```

**Comment by jpvajda** [comment]
@lukeocodes This now includes a working example of Context History with Agent.

**Comment by jpvajda** [comment]
closing as the auto generated v5 SDK now contains this. 


### **Context History Features Available:**

1. **`AgentV1Context`** class for conversation context configuration
2. **`AgentV1HistoryMessage`** class for individual history messages  
3. **`AgentV1Flags`** class with a `history` boolean flag (defaults to `True`)
4. **Full integration** in the `AgentV1SettingsMessage` and `AgentV1Agent` classes

### **Key Components Found:**

From ```87:106:/deepgram-repos/deepgram-python-sdk/src/deepgram/extensions/types/sockets/agent_v1_settings_message.py```
```python
class AgentV1HistoryMessage(UniversalBaseModel):
    """Conversation text as part of the conversation history"""
    
    type: typing.Literal["History"] = "History"
    """Message type identifier for conversation text"""
    
    role: typing.Literal["user", "assistant"]
    """Identifies who spoke the statement"""
    
    content: str
    """The actual statement that was spoken"""
```

From ```153:157:/deepgram-repos/deepgram-python-sdk/src/deepgram/extensions/types/sockets/agent_v1_settings_message.py```
```python
class AgentV1Flags(UniversalBaseModel):
    """Agent flags configuration"""
    
    history: bool = True
    """Enable or disable history message reporting"""
```

From ```169:173:/deepgram-repos/deepgram-python-sdk/src/deepgram/extensions/types/sockets/agent_v1_settings_message.py```
```python
class AgentV1Context(UniversalBaseModel):
    """Conversation context including the history of messages and function calls"""
    
    messages: typing.Optional[typing.List[typing.Union[AgentV1HistoryMessage, AgentV1HistoryFunctionCalls]]] = None
    """Conversation history as a list of messages and function calls"""
```

The new SDK supports:
- ✅ `settings.flags.history` boolean flag
- ✅ `agent.context.messages` array with `AgentV1HistoryMessage` objects  
- ✅ Message objects with `type: "History"`, `role: "user"|"assistant"`, and `content: str`
